cpp/open3d/t/geometry/TriangleMesh.cpp

```diff
@@ -1215,6 +1215,139 @@ TriangleMesh TriangleMesh::SelectByIndex(const core::Tensor &indices) const {
+    const auto vmask_type = core::Int32;
+    using vmask_itype = int32_t;
+    core::Tensor vertex_mask = core::Tensor::Zeros({orig_num_vertices}, vmask_type);
+    using Length_t = decltype(orig_num_vertices);
+    std::vector<Length_t> index_old_to_new(orig_num_vertices);
+
+    //REM_DUP_VERT_STEP 1: 
+    // Compute map from points to old indices, and use a counter
+    // to compute a map from old indices to new unique indices.
+    const Coord_t *vertices_ptr = vertices.GetDataPtr<Coord_t>();
+    vmask_itype * vertex_mask_ptr = vertex_mask.GetDataPtr<vmask_itype>();
+    Length_t k = 0;                                  // new index
+    for (Length_t i = 0; i < orig_num_vertices; ++i) {  // old index
+        Point3d coord = std::make_tuple(vertices_ptr[i * 3], vertices_ptr[i * 3 + 1],
+                                        vertices_ptr[i * 3 + 2]);
+        if (point_to_old_index.find(coord) == point_to_old_index.end()) {
+            point_to_old_index[coord] = i;
+            index_old_to_new[i] = k;
+            ++k;
+            vertex_mask_ptr[i] = 1;
+        } else {
+            index_old_to_new[i] = index_old_to_new[point_to_old_index[coord]];
+        }
+    }
+    vertex_mask = vertex_mask.To(mesh.GetDevice(), core::Bool);
+
+    //REM_DUP_VERT_STEP 2:
+    // Update vertex attributes, by shrinking them appropriately.
+    for (auto item : mesh.GetVertexAttr()) {
+        mesh.SetVertexAttr(item.first, item.second.IndexGet({vertex_mask}));
+    }
+
+    //REM_DUP_VERT_STEP 3:
+    // Remap triangle indices to new unique vertex indices, if required.
+    if (k < orig_num_vertices) {
+        core::Tensor tris = mesh.GetTriangleIndices();
+        core::Tensor tris_cpu = tris.To(core::Device()).Contiguous();
+        const int64_t num_tris = tris_cpu.GetLength();
+        //Update triangle indices.
+        Tindex_t *indices_ptr = tris_cpu.GetDataPtr<Tindex_t>();
+        for(int64_t i = 0; i < num_tris * 3; ++i) {
+            int64_t new_idx = index_old_to_new[indices_ptr[i]];
+            indices_ptr[i] = Tindex_t(new_idx);
+        }
+        //Update triangle indices, no need to update other attributes.
+        tris = tris_cpu.To(mesh.GetDevice());
+        mesh.SetTriangleIndices(tris);
+    }
+
+    utility::LogDebug(
+             "[RemoveDuplicatedVertices] {:d} vertices have been removed.",
+             (int)(orig_num_vertices - k));
+
+    return mesh;
+
+}
+
+} //Anonymous namespace
+
+///
+/// Remove duplicate vertices from a mesh and return the resulting mesh.
+///
+TriangleMesh& TriangleMesh::RemoveDuplicateVertices()
+{
+    /// This function mainly does some checks and then calls the RemoveDuplicateVerticesWorker
+    /// with appropriate data type.
+    if(!HasVertexPositions()) {
+        utility::LogWarning("TriangeMesh::RemoveDuplicateVertices: No vertices present, ignoring.");
+        return *this;
+    }
+    auto vertices = GetVertexPositions();
+    auto triangles = GetTriangleIndices();
```

Comment:
There could be meshes w/o triangles tensor. In this case, the call to `GetTrianglesIndices` throws.
Could you please add safety checks for this case?